src/cluster/src/install.rs

```diff
@@ -472,6 +478,103 @@ impl ClusterInstaller {
         let versions = helm_client.versions(DEFAULT_CHART_APP_NAME)?;
         Ok(versions)
     }
+
+    /// Get installed system chart
+    pub fn sys_charts() -> Result<Vec<InstalledChart>, ClusterError> {
+        let helm_client = HelmClient::new()?;
+        let sys_charts = helm_client.installed_sys_charts(DEFAULT_CHART_SYS_REPO)?;
+        Ok(sys_charts)
+    }
+
+    // Getting server hostname from K8 context
+    fn get_cluster_server_host(&self, kc_config: KubeContext) -> Result<String, IoError> {
+        if let Some(ctx) = kc_config.config.current_cluster() {
+            let server_url = ctx.cluster.server.to_owned();
+            let url = match Url::parse(&server_url) {
+                Ok(url) => url,
+                Err(e) => {
+                    return Err(IoError::new(
+                        ErrorKind::Other,
+                        format!("error parsing server url {}", e.to_string()),
+                    ))
+                }
+            };
+            Ok(url.host().unwrap().to_string())
+        } else {
+            Err(IoError::new(ErrorKind::Other, "no context found"))
+        }
+    }
+
+    /// Runs pre install checks
+    ///  1. Check if compatible helm version is installed
+    ///  2. Check if compatible sys charts are installed
+    ///  3. Check if the K8 config hostname is not an IP address
+    fn pre_install_check(&self) -> Result<(), ClusterError> {
+        // check helm version
+        let version_text_trimmed = self.helm_client.get_helm_version()?;
+        if Version::parse(&version_text_trimmed) < Version::parse(DEFAULT_HELM_VERSION) {
+            return Err(ClusterError::Other(format!(
+                "Helm version {} is not compatible with fluvio platform, please install version >= {}",
+                version_text_trimmed, DEFAULT_HELM_VERSION
+            )));
+        }
+
+        // check installed system chart version
+        let sys_charts = self
+            .helm_client
+            .installed_sys_charts(DEFAULT_CHART_SYS_REPO)?;
+        if sys_charts.len() == 1 {
+            let installed_chart = sys_charts.first().unwrap();
+            let installed_chart_version = installed_chart.app_version.clone();
+            // checking version of chart found
+            if Version::parse(&installed_chart_version) < Version::parse(SYS_CHART_VERSION) {
```

Comment:
@sehz, if the `app_version` is going to be empty for sys chart, how do you suggest I perform this version check?

Reply:
check only sys_version?

Reply:
So do you want to only check if the sys-chart is installed?

Reply:
I thought that's how existing code worked?

Reply:
Yeah, But we also do a version check (sys-charts > 0.2.0) For that we will need a version but the `helm list` doesn't provide version anymore.